builders/tools/walletconnect.md

```diff
@@ -0,0 +1,416 @@
+2. Run `npm install` to install the required dependencies
+3. Run `npm start` to spin up a local instance of the DApp
+
+To test the WalletConnect connection, you can use the MetaMask mobile app. For the purposes of this guide, you will need to already be connected to the Moonbase Alpha TestNet on the MetaMask mobile app. Later on in the guide, you will learn how to check if the connected network is a supported network, and if not display an error that will suggest users to switch to a network that is supported.
+
+There are a couple of ways you can connect your MetaMask mobile wallet to the Moonbase Alpha TestNet. You can manually add the Moonbase Alpha TestNet configurations from the **Networks** section of the **Settings** menu. Or you can also open up the **Browser** from MetaMask mobile and navigate to [docs.moonbeam.network](https://docs.moonbeam.network){target=blank}, click on **Connect MetaMask** at the top of the page, and select **Moonbase Alpha** from the menu. This will prompt you to automatically add Moonbase Alpha as a custom network and saves you from inputting the network configurations manually.
+
+## Connect DApp to MetaMask Mobile {: #connect-dapp-to-metamask-mobile }
+
+In this section, you will learn how to make a connection between your DApp and MetaMask mobile. WalletConnect establishes a remote connection between a DApp and mobile wallet by using a bridge server to relay payloads. The connection is initiated via a QR code displayed in the DApp, which will need to be scanned and approved by the mobile wallet.
+
+To get started, you can open up the `App.js` file of the template and the first changes will be made within the `connect` function. This function will handle the connection logic by creating a new instance of the WalletConnect connector. You'll notice that the `setFetching` state hook is already in place. This will be used to set the `fetching` state variable to `true` while the connection is being established. In general the `connect` function will:
+
+1. Create the WalletConnect Connector and pass in the URL for the bridge server and the WalletConnect QR code modal
+2. Use the `setConnector` state hook to update the `connector` state variable
+3. Check if the connection has been established, and if not create a new session request
+
+```js
+const connect = async () => { 
+  setFetching(true);
+
+  // 1. Create connector
+  const connector = new WalletConnect({ bridge: "https://bridge.walletconnect.org", qrcodeModal: QRCodeModal });
+
+  // 2. Update the connector state
+  setConnector(connector);
+
+  // 3. If not connected, create a new session
+  if (!connector.connected) {
+    await connector.createSession();
+  }
+};
+```
+
+Now that you have the `connect` function setup, you can create a **Connect Wallet** button that will call it `onClick`. You can replace the `{/* buttons and network details will go here */}` comment with the following button:
```

Comment:
Where is this `buttons and network details will go here` ?

Reply:
it's from the template... I'll make it more clear in the guide